di_website/contactus/models.py

```diff
@@ -0,0 +1,131 @@
+from django.db import models
+from django.shortcuts import render
+
+from modelcluster.fields import ParentalKey
+from wagtail.admin.edit_handlers import (
+    FieldPanel, FieldRowPanel,
+    InlinePanel, MultiFieldPanel
+)
+from wagtail.core.fields import RichTextField
+from wagtail.contrib.forms.models import AbstractEmailForm, AbstractFormField,AbstractFormSubmission
+from di_website.common.mixins import HeroMixin
+from di_website.common.base import hero_panels
+from di_website.contactus.utils import CustomFormBuilder
+
+import json
+from django.core.serializers.json import DjangoJSONEncoder
+
+HONEYPOT_FORM_FIELD = 'captcha'
+
+
+class FormFields(AbstractFormField): 
+    page = ParentalKey('ContactPage', on_delete=models.CASCADE, related_name='form_fields')
+
+    """
+        Form fields with additional placeholder text that describes what the form fields are meant for
+    """
+    placeholder = models.CharField(
+        blank=True,
+        max_length=255,
+        help_text='Optional placeholder for the field'
+    )
+
+    panels = AbstractFormField.panels + [
+        FieldPanel('placeholder'),
+    ]
+
+"""
+    Contact Us form, to inherit the emailing functionality from wagtail, we'll use AbstractEmailForm
+"""
+class ContactPage(HeroMixin,AbstractEmailForm):
+
+    template = "contactus/contact_page.html"
+
+    form_fields = CustomFormBuilder
+
+    intro = RichTextField(blank=True)
+    success_alert = models.CharField(
+        max_length=255,
+        default='Your message was sent successfully',
+        )
+
+
+    content_panels = AbstractEmailForm.content_panels + [
+        hero_panels(),
+        FieldPanel('intro', classname="full"),
+        InlinePanel('form_fields', label="Form fields"),
+        
+        FieldPanel('success_alert', classname="full"),
+        MultiFieldPanel([
+            FieldRowPanel([
+                FieldPanel('from_address', classname="col6"),
+                FieldPanel('to_address', classname="col6"),
+            ]),
+            FieldPanel('subject'),
+        ], "Email"),
+    ]
+
+    def process_form_submission(self, form):
+       # form_data = json.dumps(form.cleaned_data, cls=DjangoJSONEncoder)
+        self.get_submission_class().objects.create(
+            form_data=json.dumps(form.cleaned_data, cls=DjangoJSONEncoder),
+            page=self,
+        )
+        return form.cleaned_data
+
+    def get_placeholder_for_field(self, label):
+        try:
+            return self.form_fields.filter(label=label).first().placeholder
+        except Exception:
```

Comment:
KeyError? IndexError? You can also combine multiple possible exceptions with tibble `except (KeyError, IndexError) as e:`